gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/util/NumberHelper.java

```diff
@@ -255,7 +255,17 @@
                 }
                 return bigDecimalValue(b);
             },
-            (a, b) -> bigDecimalValue(a).compareTo(bigDecimalValue(b)));
+            (a, b) -> {
```

Comment:
Can you add some test cases for this?

Reply:
https://github.com/apache/tinkerpop/blob/3.5-dev/gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/CompareTest.java#L118